metrics/web/lib/common/presentation/routes/route_generator.dart

```diff
@@ -1,20 +1,23 @@
 import 'package:flutter/material.dart';
 import 'package:metrics/auth/presentation/pages/login_page.dart';
+import 'package:metrics/common/presentation/pages/loading_page.dart';
 import 'package:metrics/common/presentation/routes/metrics_page_route.dart';
 import 'package:metrics/common/presentation/routes/route_name.dart';
-import 'package:metrics/common/presentation/pages/loading_page.dart';
 import 'package:metrics/dashboard/presentation/pages/dashboard_page.dart';
+import 'package:metrics/debug_menu/presentation/pages/debug_menu_page.dart';
 import 'package:metrics/project_groups/presentation/pages/project_group_page.dart';
 
 /// Responsible for generating routes.
 class RouteGenerator {
   /// Generates a route for the given route [settings]
-  /// based on the [isLoggedIn] authentication status.
+  /// based on the [isLoggedIn] authentication status and [isDebugMenuEnabled]
+  /// feature status.
   ///
   /// Throws an [AssertionError] if [settings] is null.
   static MetricsPageRoute generateRoute({
     @required RouteSettings settings,
     bool isLoggedIn,
+    bool isDebugMenuEnabled,
```

Comment:
Let's display a message on the debug menu page that the debug feature is disabled if a user navigates directly to it. Adding one more condition to navigation makes it too complex and can affect new navigation system integration.